bin/strap.sh

```diff
@@ -254,156 +308,141 @@ fi
+    HOMEBREW_PREFIX="/opt/homebrew"
+    HOMEBREW_REPOSITORY="${HOMEBREW_PREFIX}"
+  else
+    HOMEBREW_PREFIX="/usr/local"
+    HOMEBREW_REPOSITORY="${HOMEBREW_PREFIX}/Homebrew"
   fi
 fi
+[ -d "$HOMEBREW_PREFIX" ] || sudo_askpass mkdir -p "$HOMEBREW_PREFIX"
+if [ "$HOMEBREW_PREFIX" = "/usr/local" ]; then
+  sudo_askpass chown "root:wheel" "$HOMEBREW_PREFIX" 2>/dev/null || true
+fi
+(
+  cd "$HOMEBREW_PREFIX"
+  sudo_askpass mkdir -p Cellar Caskroom Frameworks bin etc include lib opt sbin share var
+  sudo_askpass chown "$USER:admin" Cellar Caskroom Frameworks bin etc include lib opt sbin share var
+)
 
-logk
+[ -d "$HOMEBREW_REPOSITORY" ] || sudo_askpass mkdir -p "$HOMEBREW_REPOSITORY"
+sudo_askpass chown -R "$USER:admin" "$HOMEBREW_REPOSITORY"
 
-# Setup Homebrew directory and permissions.
-logn "Check for Homebrew..."
-if ! type brew 1>/dev/null 2>&1 ; then
-  logdebug "Installing Homebrew"
-  BREW_INSTALL_SCRIPT="/tmp/brew-install.rb"
-  curl -fsSL --output $BREW_INSTALL_SCRIPT https://raw.githubusercontent.com/Homebrew/install/master/install
-  /usr/bin/ruby $BREW_INSTALL_SCRIPT
+if [ $HOMEBREW_PREFIX != $HOMEBREW_REPOSITORY ]; then
+  ln -sf "$HOMEBREW_REPOSITORY/bin/brew" "$HOMEBREW_PREFIX/bin/brew"
 fi
-logk
-
-export PATH="$HOMEBREW_PREFIX/bin:$PATH"
 
-# Unshallow old Homebrew taps
-for d in 'homebrew/core' 'homebrew/cask' 'homebrew/services' 'daptiv/homebrew-tap'; do
-  HOMEBREW_REPO="$(brew --repo $d)"
-  if [ -d "$HOMEBREW_REPO" ] && [ "$(git -C "$HOMEBREW_REPO" rev-parse --is-shallow-repository)" = 'true' ]; then
-    logn "Unshallowing $HOMEBREW_REPO..."
-    git -C "$HOMEBREW_REPO" fetch --unshallow
-    logk
-  else
-    logdebug "$HOMEBREW_REPO does not exist or doesn't need to be unshallowed."
-  fi
-done
+# Download Homebrew.
+export GIT_DIR="$HOMEBREW_REPOSITORY/.git" GIT_WORK_TREE="$HOMEBREW_REPOSITORY"
+git init $Q
+git config remote.origin.url "https://github.com/Homebrew/brew"
+git config remote.origin.fetch "+refs/heads/*:refs/remotes/origin/*"
+git fetch $Q --tags --force
+git reset $Q --hard origin/master
+unset GIT_DIR GIT_WORK_TREE
+logk
 
 # Update Homebrew.
-logn "Updating Homebrew..."
-if [ -n "$STRAP_DEBUG" ]; then
-  brew update
-else
-  brew update 1>/dev/null 2>&1
-fi
+export PATH="$HOMEBREW_PREFIX/bin:$PATH"
+logn "Updating Homebrew:"
+brew update --quiet
 logk
 
 # Install Homebrew Bundle, Cask and Services tap.
-logn "Installing Homebrew taps and extensions..."
-if [ -n "$STRAP_DEBUG" ]; then
-brew bundle --file=- <<EOF
-tap 'homebrew/cask'
-tap 'homebrew/core'
-tap 'homebrew/services'
-tap 'daptiv/homebrew-tap'
-EOF
-else
-brew bundle --file=- 1>/dev/null 2>&1 <<EOF
-tap 'homebrew/cask'
-tap 'homebrew/core'
-tap 'homebrew/services'
-tap 'daptiv/homebrew-tap'
-EOF
-fi
+log "Installing Homebrew taps and extensions:"
+brew bundle --quiet --file=- <<RUBY
+tap "homebrew/cask"
+tap "homebrew/core"
+tap "homebrew/services"
+RUBY
 logk
 
-if (( "${BASH_VERSINFO[0]}" < 5 )); then
```

Comment:
If we are moving everything to zsh, it is fine to remove this from here.  But I would still like to have bash upgraded in our dotfiles repo.

Reply:
this is in PR 40